server/PythonClient/Client.py

```diff
@@ -0,0 +1,339 @@
+        self.socket.connect((host, port))
+
+    def getSendAllFunction(self):
+        return SendFunction(self.socket.sendall)
+
+    def getReciever(self):
+        return self.socket.recv
+        pass
+
+
+# CONNECT TO SERVER
+client = FractalClient(HOST, PORT)
+
+reciever = client.getReciever()
+UserId = NewType('UserId', int)
+
+
+class StreamHelper:
+
+    byteOrder = "big"  # big or little
+
+    @staticmethod
+    def readByte(reader: RecieveFunction, signed: bool = False):
+        return int.from_bytes(
+            reader(1), byteorder=StreamHelper.byteOrder, signed=signed)
+
+    @staticmethod
+    def readShort(reader: RecieveFunction, signed: bool = False):
+        return int.from_bytes(
+            reader(2), byteorder=StreamHelper.byteOrder, signed=signed)
+        pass
+
+    @staticmethod
+    def readInt(reader: RecieveFunction, signed: bool = False):
+        return int.from_bytes(
+            reader(4), byteorder=StreamHelper.byteOrder, signed=signed)
+        pass
+
+    @staticmethod
+    def readLong(reader: RecieveFunction, signed: bool = False):
+        return int.from_bytes(
+            reader(8), byteorder=StreamHelper.byteOrder, signed=signed)
+        pass
+
+    @staticmethod
+    def readFloat(reader: RecieveFunction):
+        struct.unpack('f', reader(4))
+
+    @staticmethod
+    def readDouble(reader: RecieveFunction):
+        struct.unpack('f', reader(8))
+
+    @staticmethod
+    def readBytesToString(reader: RecieveFunction, nBytes):
+        return reader(nBytes).decode(ENCODING)
+
+# RESPONSIBLE FOR READING DATA FROM SERVER
+
+
+class FractalReader:
+
+    def __init__(self):
+        super().__init__()
+
+    def read(self, reciever: RecieveFunction):
+        # Length of paylaod in bytes
+        payloadlen = StreamHelper.readInt(reciever)
+
+        # Read payload name length
+        payloadNameLen = StreamHelper.readInt(reciever)
+        # Read payload name
+        payloadName = StreamHelper.readBytesToString(
+            reciever, payloadNameLen)
+
+        # Read segments length
+        segmentsLen = StreamHelper.readInt(reciever)
+        # Read segments
+        segments = StreamHelper.readBytesToString(
+            reciever, segmentsLen)
+
+        # Read JSON payload length
+        jsonLen = StreamHelper.readInt(reciever)
+        # Read JSON payload
+        jsondata = StreamHelper.readBytesToString(
+            reciever, jsonLen)
+
+        # parse JSON payload data
+        payloadData = json.loads(jsondata)
+
+        # parse segment JSON payload data
+        parsed_segments = json.loads(segments)
+
+        return RecievedPayload(payloadData, self.readSegments(reciever, parsed_segments))
+
+    def readSegments(self, reciever, parsed_segments):
+        segmentDict = {}
+        for segment in parsed_segments:
+            segmentKey = next(iter(segment.keys()))
+            segmentMeta = next(iter(segment.values()))
```

Comment:
nå grun til at du ikke bare
```suggestion
            segmentKey = segment.keys()[0]
            segmentMeta = segment.values()[0]
```

Reply:
Er måten folk siere det gjøres på ellers må man først caste til Liste som er unødvendig :P